bytecode/cfg.py

```diff
@@ -61,32 +70,23 @@ def _compute_stack_size(block, size, maxsize):
         if isinstance(instr, SetLineno):
             continue
 
-        size += instr.stack_effect
-        maxsize = max(maxsize, size)
-
-        if size < 0:
-            msg = 'Failed to compute stacksize, got negative size'
-            raise RuntimeError(msg)
-
         if instr.has_jump():
+            # first compute the taken-jump path
+            before_size = size
+            update_size(instr.stack_effect(jump=True))
             target_size = size
```

Comment:
Since we do not need anymore to update target_size to handle some special cases, I think we can completely remove it.

If we get rid off `nonlocal`, we could remove `before_size` and do 
```python
target_size, maxsize = update_size(instr.stack_effect(jump=True))```

Reply:
indeed.  i've made some other simplifications too